source/core/shape/ovus.cpp

```diff
@@ -883,46 +896,59 @@ void Ovus::UVCoord(Vector2d& Result, const Intersection *Inter, TraceThreadData
 *
 * CHANGES
 *
+* due to the addition of distance, the mapping is changed to something similar
+* to lemon, cone & cylinder
+*
 ******************************************************************************/
 
 void Ovus::CalcUV(const Vector3d& IPoint, Vector2d& Result) const
 {
-    DBL len, x, y, z;
+    DBL len, x, z, t;
     DBL phi, theta;
     Vector3d P;
 
-    // Transform the ray into the ovus space.
+    // Transform the point back into the ovus space.
     MInvTransPoint(P, IPoint, Trans);
 
-    // the center of UV coordinate is the bottom center when top radius ->0
-    // and it is the top center when top radius -> 2.0* bottom radius
-    // when top radius == bottom radius, it is half-way between both center
-    //
-    // bottom center is <0,0,0>
-    // top center is <0,BottomRadius,0>
-    // TODO FIXME - comment doesn't seem to match the following code
+    // the center of UV coordinate is the <0,0> point
     x = P[X];
-//  y = P[Y] - BottomRadius*(TopRadius/(2.0*BottomRadius));
-    y = P[Y] - (TopRadius/2.0);
     z = P[Z];
 
-    // now assume it's just a sphere, for UV mapping/projection
-    len = sqrt(x * x + y * y + z * z);
+    // Determine its angle from the point (0, 0, 0) in the x-z plane.
+    len = x * x + z * z;
 
-    if (len == 0.0)
-        return;
+    if ((P[Y]>EPSILON)&&(P[Y]<(VerticalSpherePosition-EPSILON)))
+    {
+    // when not on a face, the range 0.25 to 0.75 is used (just plain magic 25% for face, no other reason, but it makes C-Lipka happy)
+        phi = 0.75-0.5*(P[Y])/(VerticalSpherePosition);
+    }
+    else if (P[Y]>EPSILON)
+    {
+    // aka P[Y] is above VerticalSpherePositon, use TopRadius, from 0% to 25%
+       phi = 0.0;
```

Comment:
Please stick to a 4-blanks indent.
Also, please indent the comment inside the braces.